packages/astro/src/vite-plugin-css/index.ts

```diff
@@ -24,6 +29,43 @@ function getComponentFromVirtualModuleCssName(virtualModulePrefix: string, id: s
 		.replace(new RegExp(ASTRO_CSS_EXTENSION_POST_PATTERN, 'g'), '.');
 }
 
+/**
+ * Walk down the dependency tree to collect CSS with depth/order.
+ * Performs depth-first traversal to ensure correct CSS ordering based on import order.
+ */
+function* collectCSSWithOrder(
+	id: string,
+	mod: vite.EnvironmentModuleNode,
+	seen = new Set<string>(),
+): Generator<ImportedDevStyle & { id: string; idKey: string }, void, unknown> {
+	seen.add(id);
+
+	// Keep all of the imported modules into an array so we can go through them one at a time
+	const imported = Array.from(mod.importedModules);
+
+	// Check if this module is CSS and should be collected
+	if (isBuildableCSSRequest(id)) {
+		yield {
+			id,
+			idKey: id,
+			content: '',
+			url: wrapId(id),
+		};
+		return;
+	}
+	// ?raw imports the underlying css but is handled as a string in the JS.
+	else if(id.endsWith('?raw')) {
+		return;
+	}
+
+	// Recursively walk imported modules (depth-first)
+	for (const imp of imported) {
+		if (imp.id && !seen.has(imp?.id)) {
+			yield* collectCSSWithOrder(imp.id, imp, seen);
+		}
+	}
```

Comment:
Nit: I think if we avoid the recursion and we just use a simple loop, it will get faster (debatable) and use less memory